platform/publisher.go

```diff
@@ -23,46 +23,47 @@
 package platform
 
 import (
-	"crypto/md5"
-	"encoding/hex"
+	"os"
+
 	"github.com/JetBrains/qodana-cli/v2025/cloud"
 	"github.com/JetBrains/qodana-cli/v2025/platform/qdenv"
 	"github.com/JetBrains/qodana-cli/v2025/platform/utils"
+	"github.com/JetBrains/qodana-cli/v2025/tooling"
 	log "github.com/sirupsen/logrus"
-	"io"
-	"net/http"
-	"os"
-	"path/filepath"
 )
 
-const PublisherJarName = "publisher-cli.jar"
-const PublisherVersion = "2.1.31"
-
 type Publisher struct {
 	ResultsDir string
 	LogDir     string
 	AnalysisId string
 }
 
 // SendReport sends report to Qodana Cloud.
-func SendReport(publisher Publisher, token string, publisherPath string, javaPath string) {
-	if _, err := os.Stat(publisherPath); os.IsNotExist(err) {
-		err := os.MkdirAll(filepath.Dir(publisherPath), os.ModePerm)
-		if err != nil {
-			log.Fatalf("failed to create directory: %v", err)
-		}
-		fetchPublisher(publisherPath)
+func SendReport(publisher Publisher, token string, javaPath string) {
+	fp, err := os.CreateTemp("", "qodana-publisher")
+	if err != nil {
+		log.Fatalf("Failed to create a temporary file: %s", err)
 	}
-	if _, err := os.Stat(publisherPath); os.IsNotExist(err) {
-		log.Fatalf("Not able to send the report: %s is missing", publisherPath)
+	publisherPath := fp.Name()
+	err = fp.Close()
+	if err != nil {
+		log.Fatalf("Failed to close temporary file %q: %s", fp.Name(), err)
 	}
+	defer func() {
+		err = os.Remove(fp.Name())
+		if err != nil {
+			log.Fatalf("Failed to remove temporary file %q: %s", fp.Name(), err)
+		}
+	}()
+
+	extractPublisher(publisherPath)
 
 	publisherCommand := getPublisherArgs(
 		javaPath,
 		publisherPath,
 		publisher,
 		token,
-		cloud.GetCloudApiEndpoints().CloudApiUrl,
+		cloud.DefaultEndpoint,
```

Comment:
I believe we have missed here the custom cloud endpoint, for self hosted the endpoint could be different

Reply:
Fixed, I think